File: update_captions.js
const fs = require('fs');
let content = fs.readFileSync('index.html', 'utf-8');

content = content.replace(/(<img[^>]*?src="([^"]+)"[^>]*>\s*)(<div class="fc-slide-caption">)(.*?)(<\/div>)/gs, (match, p1, p2, p3, p4, p5) => {
    let filename = p2.split('/').pop().replace(/\.[^/.]+$/, "");
    return p1 + p3 + filename + p5;
});

fs.writeFileSync('index.html', content, 'utf-8');
console.log('Updated captions based on filenames.');

File: update_captions.py
import re, os

filepath = "index.html"
with open(filepath, "r", encoding="utf-8") as f:
    content = f.read()

def repl(match):
    img_tag = match.group(1)
    src_val = match.group(2)
    filename = src_val.split('/')[-1].rsplit('.', 1)[0]
    caption_start = match.group(3)
    caption_end = match.group(4)
    return f"{img_tag}{caption_start}{filename}{caption_end}"

pattern = re.compile(r'(<img[^>]*?src="([^"]+)"[^>]*>\s*)(<div class="fc-slide-caption">)(.*?)(</div>)', re.DOTALL)
new_content = pattern.sub(repl, content)

with open(filepath, "w", encoding="utf-8") as f:
    f.write(new_content)
print("Updated captions based on filenames.")
